tests/integration/ovs_test.py

```diff
@@ -15,106 +14,102 @@
-"""
+@pytest.mark.xfail(
+    raises=NmstateLibnmError, reason='https://bugzilla.redhat.com/1724901'
+)
+def test_create_and_remove_ovs_bridge_with_min_desired_state():
+    with ovs_bridge('br1') as bridge_state:
+        assertlib.assert_state_match(bridge_state)
+    assertlib.assert_absent('br1')
 
 
 @pytest.mark.xfail(
     raises=NmstateLibnmError, reason='https://bugzilla.redhat.com/1724901'
 )
-def test_create_and_remove_ovs_bridge_with_a_system_port(eth1_up):
-    state = yaml.load(OVS_BRIDGE_YAML_BASE, Loader=yaml.SafeLoader)
-    state[INTERFACES][0]['bridge']['port'] = [
-        {'name': 'eth1', 'type': 'system'}
-    ]
-    libnmstate.apply(state)
-
-    setup_remove_ovs_bridge_state = {
-        INTERFACES: [
-            {'name': 'ovs-br0', 'type': 'ovs-bridge', 'state': 'absent'}
-        ]
-    }
-    libnmstate.apply(setup_remove_ovs_bridge_state)
-    state = statelib.show_only((state[INTERFACES][0]['name'],))
-    assert not state[INTERFACES]
+def test_create_and_remove_ovs_bridge_options_specified():
+    bridge = create_bridge_subtree_state(
+        {
+            OVSBridge.FAIL_MODE: '',
+            OVSBridge.MCAST_SNOOPING_ENABLED: False,
+            OVSBridge.RSTP: False,
+            OVSBridge.STP: True,
+        }
+    )
+
+    with ovs_bridge('br1', bridge) as bridge_state:
+        assertlib.assert_state_match(bridge_state)
+
+    assertlib.assert_absent('br1')
 
 
 @pytest.mark.xfail(
     raises=NmstateLibnmError, reason='https://bugzilla.redhat.com/1724901'
 )
-def test_create_and_remove_ovs_bridge_with_min_desired_state():
-    desired_state = {
-        INTERFACES: [{'name': 'ovs-br0', 'type': 'ovs-bridge', 'state': 'up'}]
-    }
-    libnmstate.apply(desired_state)
+def test_create_and_remove_ovs_bridge_with_a_system_port(eth1_up):
+    bridge = create_bridge_subtree_state()
+    add_port_to_bridge(bridge, 'eth1', OVSBridgePortType.SYSTEM)
 
-    setup_remove_ovs_bridge_state = {
-        INTERFACES: [
-            {'name': 'ovs-br0', 'type': 'ovs-bridge', 'state': 'absent'}
-        ]
-    }
-    libnmstate.apply(setup_remove_ovs_bridge_state)
-    state = statelib.show_only((desired_state[INTERFACES][0]['name'],))
-    assert not state[INTERFACES]
+    with ovs_bridge('br1', bridge) as bridge_state:
+        assertlib.assert_state_match(bridge_state)
+
+    assertlib.assert_absent('br1')
 
 
 @pytest.mark.xfail(
     raises=NmstateLibnmError, reason='https://bugzilla.redhat.com/1724901'
 )
 def test_create_and_remove_ovs_bridge_with_an_internal_port():
-    state = yaml.load(OVS_BRIDGE_YAML_BASE, Loader=yaml.SafeLoader)
-    state[INTERFACES][0]['bridge']['port'] = [
-        {'name': 'ovs0', 'type': 'internal'}
-    ]
-    ovs_internal_interface_state = {
-        'name': 'ovs0',
-        'type': 'ovs-interface',
-        'state': 'up',
-        'mtu': 1500,
-        'ipv4': {
+    bridge = create_bridge_subtree_state()
+    add_port_to_bridge(bridge, 'ovs1', OVSBridgePortType.INTERNAL)
```

Comment:
A constant or a local constant for ‘ovs1’ will be nice.

Reply:
Because you re-use it all over in the test, and defining the name once is better over repeating. It is also clear that writer intended for the same thing and not picked a random name.

Reply:
Done